packages/core/src/services/chains/ChainValidator/index.ts

```diff
@@ -48,14 +63,41 @@ export class ChainValidator {
 
     if (providerResult.error) return providerResult
 
+    const provider = providerResult.unwrap()
+
+    const freeQuota = await this.checkFreeProviderUsageQuota(provider)
+    if (freeQuota.error) return freeQuota
```

Comment:
we are unwraping in line 66 and returning a result error here instead, what is the correct behaviour for error handling here?

Reply:
My idea is that `ChainValidator` return Result and is at the last moment on the `runChain` that unwrap the error if any. Otherwise means we have a healthty chain step

Reply:
but you are throwing an error instead of returning a result error in line 66

Reply:
That's wrong, chainging

Reply:
Ah not wait. Check prev line

Reply:
.value instead of unwrap if this is an ok result then please